assets/react/v3/entries/course-builder/components/layouts/Notebook.tsx

```diff
@@ -0,0 +1,366 @@
+import { styleUtils } from '@Utils/style-utils';
+import Show from '@Controls/Show';
+
+import { type TResizeType, useResize } from '@Hooks/useResize';
+import { getFromLocalStorage, setToLocalStorage } from '@Utils/localstorage';
+import { LocalStorageKeys, notebook } from '@Config/constants';
+
+interface Position {
+	x: number;
+	y: number;
+}
+
+interface NotebookData {
+	position: {
+		left: string;
+		top: string;
+		height: string;
+		width: string;
+	};
+	notes: string;
+}
+
+const Notebook = () => {
+	const [isCollapsed, setIsCollapsed] = useState<boolean>(true);
+	const [isDragging, setIsDragging] = useState<boolean>(false);
+	const [isFloating, setIsFloating] = useState<boolean>(false);
+	const [offset, setOffset] = useState<Position>({ x: 0, y: 0 });
+	const [contentEditable, setContentEditable] = useState<boolean>(false);
+	const [content, setContent] = useState<string>('');
+
+	const wrapperRef = useRef<HTMLDivElement>(null);
+
+	const expandAnimation = useSpring({
+		height: !isCollapsed ? notebook.MIN_NOTEBOOK_HEIGHT : notebook.NOTEBOOK_HEADER,
+		config: {
+			duration: 300,
+			easing: (t) => t * (2 - t),
+		},
+	});
+
+	const { handleResize, isResizing, size } = useResize({
+		resizeDivRef: wrapperRef,
+		options: {
+			minHeight: notebook.MIN_NOTEBOOK_HEIGHT,
+			minWidth: notebook.MIN_NOTEBOOK_WIDTH,
+		},
+	});
+
+	const onContentBlur = (event: React.FocusEvent) => {
+		setContentEditable(false);
+		const notebookData: NotebookData = JSON.parse(getFromLocalStorage(LocalStorageKeys.notebook) || '{}');
+		const tempDiv = document.createElement('div');
+		tempDiv.innerHTML = event.currentTarget.innerHTML;
+		const text = tempDiv.innerText || '';
+		setToLocalStorage(
+			LocalStorageKeys.notebook,
+			JSON.stringify({
+				...notebookData,
+				notes: text,
+			})
+		);
+	};
+
+	const handleMouseDown = (event: React.MouseEvent) => {
+		event.preventDefault();
+		event.stopPropagation();
+
+		if (isCollapsed) return;
+
+		setIsDragging(true);
+		const { clientX, clientY } = event;
+		const wrapper = wrapperRef.current;
+
+		if (wrapper) {
+			const rect = wrapper.getBoundingClientRect();
+			setOffset({ x: clientX - rect.left, y: clientY - rect.top });
+		}
+	};
+
+	useEffect(() => {
+		const handleMouseMove = throttle((event: MouseEvent) => {
+			if (!isDragging || !isDefined(wrapperRef.current)) return;
+			setIsFloating(true);
+
+			const warpper = wrapperRef.current;
+			const { offsetWidth: notebookWidth, offsetHeight: notebookHeight } = warpper;
+			const { innerWidth: windowWidth, innerHeight: windowHeight } = window;
+
+			const newX = Math.min(Math.max(event.clientX - offset.x, 0), windowWidth - notebookWidth);
+			const newY = Math.min(Math.max(event.clientY - offset.y, 0), windowHeight - notebookHeight);
+
+			warpper.style.left = `${newX}px`;
+			warpper.style.top = `${newY}px`;
+		}, 10);
+
+		const handleMouseUp = () => {
+			setIsDragging(false);
+
+			if (isDefined(wrapperRef.current)) {
```

Comment:
Use guard clause instead.
```suggestion
			if (!isDefined(wrapperRef.current)) {
			  return;
			}
```